### DISCUSSION TITLE
[Enhancement] Preventing lag with huge libraries

### DISCUSSION THREAD

**Discussion by neumesounds** [ciderapp/Cider#866] [Feature Request]
Whenever I hit the songs tab, I'm met with a 5-10 second wait before it changes to that tab. Scrolling through the Recently Added bit is scroll, wait for it to process, scroll again, wait for it to process. I have ram to spare and 1-2gb while running isn't the end of the world but but having 11,500 songs is taking a toll on how fast this runs on Windows, Linux, and MacOS. Perhaps it could be chunked to 1000 tiles and get more when needed?

**Comment by cryptofyre** [comment]
Pretty sure we can increase the chunk size albeit increasing CPU usage for a duration of time and its mostly to prevent API rate limiting. 

I’ll rope in @booploops to comment.
